cmake/modules/AddSwift.cmake

```diff
@@ -536,8 +535,29 @@ function(add_swift_host_library name)
         "LINKER:-current_version,${SWIFT_COMPILER_VERSION}")
     endif()
 
+    # Always link as CXX even if we have swift content unless we only contain
+    # swift content signaled via us being marked "PURE_SWIFT".
+    if (NOT ASHL_PURE_SWIFT)
+      set_target_properties(${name} PROPERTIES LINKER_LANGUAGE CXX)
+
+      # If we found a swift compiler and are going to use swift code in swift
+      # host side tools but link with clang, add the appropriate -L paths so we
+      # find all of the necessary swift libraries.
+      if (CMAKE_Swift_COMPILER)
+        # Add in the SDK directory for the host platform and add an rpath.
+        target_link_libraries(${name} PRIVATE
+          -L${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_ARCH_${SWIFT_HOST_VARIANT_ARCH}_PATH}/usr/lib/swift)
```

Comment:
This can be incompatible with the version of the Swift compiler that is in use no?

Reply:
I don't understand, can you elaborate?

Reply:
Not all platforms have ABI stability.  There is no guarantee that the libraries being used are going to be compatible.  I think that we should not be using the just built resources unless we can guarantee that the Swift compiler is the just built compiler.

Reply:
That is not what is happening here. What is happening here is that I am inserting the path to the SDK that is associated with the host platform. Which is already what we would be using to compile the host side of things. That toolchain also has a compatibility dir that I need.

Reply:
I talked with compnerd. The real issue is I forgot to conditionalize this on being on Darwin. :doh:.

Reply:
Actually I did conditionalize it already. But some other parts of this I should have unconditionalized.